### PR TITLE
games-fps/gzdoom: Bump to version 4.2.0

### PR DESCRIPTION
**Pull request by vilhelmgray**
The 'nonfree' USE flag is introduced to allow users to control whether
nonfree assets are installed.

As well, the license information is updated to account for nonfree assets.

Doom assets are under the Doom Collector's Edition license. Heretic and
Hexen assets are under the Activision license. Chex Quest assets are
under the Chex Quest 3 license. Strife assets are freely distributable
according to Stephen Kick, CEO of Nightdive Studios.

### CODE REVIEW

File: games-fps/freedm/freedm-0.11.3-r1.ebuild [hunk 1]
```diff
@@ -13,7 +13,7 @@ KEYWORDS="~amd64 ~x86"
 RDEPEND="
 	~games-fps/freedm-data-${PV}
 	|| (
-		games-fps/gzdoom
+		games-fps/gzdoom[nonfree]
```
Comment (chewi) [inline games-fps/freedm/freedm-0.11.3-r1.ebuild]: You can avoid forcing the gzdoom update by changing this to `games-fps/gzdoom[nonfree(+)]`.

File: games-fps/freedoom/freedoom-0.11.3-r2.ebuild [hunk 2]
```diff
@@ -13,7 +13,7 @@ KEYWORDS="~amd64 ~x86"
 RDEPEND="
 	~games-fps/freedoom-data-${PV}
 	|| (
-		games-fps/gzdoom
+		games-fps/gzdoom[nonfree]
```
Comment (chewi) [inline games-fps/freedoom/freedoom-0.11.3-r2.ebuild]: Same as above.

**Comment by vilhelmgray** [comment]
> You don't appear to have submitted these non-free patches upstream. Do you think they wouldn't be interested?

The patch to control whether non-free assets are built was submitted here: https://github.com/coelckers/gzdoom/pull/923

Unfortunately, it was rejected by the GZDoom project lead Christoph Oelckers. It's understandable since the non-free assets are required in order for GZDoom to run the popular supported game titles, and the primary reason most users download a Doom engine is to play the eponymous game.

However, as another GZDoom developer Rachael Alexanderson points out, GZDoom is capable of running other games without the non-free assets, so I've decided to keep this patch for the Gentoo build so that users have the freedom to choose.

As for the status of the other two patches, the [fluidsynth patch was merged after the 4.2.0 release](https://github.com/coelckers/gzdoom/pull/865), while the soundfont installation patch is still pending (https://github.com/coelckers/gzdoom/pull/874).

**Comment by chewi** [comment]
Thanks for the thorough explanation!

**Pull request closed without merging**